integrations/webchat/examples/pre-post-chat-forms/client/javascript/index.html

```diff
@@ -277,121 +286,123 @@
+    });
+  }
+
+  /**
+   * This handler is called before web chat sends a message to the assistant. We will use this to send the name the
+   * user entered when the welcome node is requested.
+   */
+  function preSendHandler(event) {
+    // This code only runs for the initial welcome message, but you could set the variable on any message that is sent
+    // to the assistant.
+    if (event.data.history && event.data.history.is_welcome_request) {
+      // Make sure these objects exist but don't override them if they already do.
+      // Note: If you are using a dialog skill instead of an actions skill, then replace "actions skill" with
+      // "main skill" and replace "skill_variables" with "user_defined".
+      event.data.context.skills['actions skill'] = event.data.context.skills['actions skill'] || {};
+      event.data.context.skills['actions skill'].skill_variables = event.data.context.skills['actions skill'].skill_variables || {};
+
+      // Set the name that was previously entered by the user.
+      event.data.context.skills['actions skill'].skill_variables.User_Name = fullName;
+    } else {
+      // Set this flag for a non-welcome message so we can now show the post-chat form if the user closes the web
+      // chat.
+      hasSentMessage = true;
     }
-
-    // This is the standard web chat configuration object. You can modify these values with the embed code for your
-    // own assistant if you wish to try this example with your assistant. You can find the documentation for this at
-    // https://web-chat.global.assistant.watson.cloud.ibm.com/docs.html?to=api-configuration#configurationobject.
-    window.watsonAssistantChatOptions = {
-      integrationID: "a03df684-bee7-42d3-8d87-1db9c8e51bec",
-      region: "us-south",
-      serviceInstanceID: "9a3613d2-3ce6-4928-8eb6-4d659d87ae68",
-      onLoad: onLoad,
-    };
-    setTimeout(function(){const t=document.createElement('script');t.src="https://web-chat.global.assistant.watson.appdomain.cloud/versions/" + (window.watsonAssistantChatOptions.clientVersion || 'latest') + "/WatsonAssistantChatEntry.js";document.head.appendChild(t);});
-  </script>
+  }
+
+  /**
+   * This is the function that is called when the web chat code has been loaded and it is ready to be rendered.
+   */
+  function onLoad(instance) {
+    // Add listeners that will use a custom panel to display custom content when web chat is opened or closed. You
+    // can use view:change instead of view:pre:change but view:pre:change gives you the option of including a
+    // cancel button to allow the user to keep web chat open. You can also use the agent start and end chat events
+    // to accomplish the same thing when users are interacting with a human agent from a service desk. Using
+    // "view:change" allows us to display a custom panel after the web chat window has been opened; with
+    // "view:pre:change" web chat will not be open yet and you can't show a custom panel if web chat is closed.
```

Comment:
Part of this comment seems to be referring to the pre-chat form behavior and part is referring to the post-chat form behavior. It may be easier to understand if you're explicit about that in the comment.

Also, this seems to contradict
>You can use view:change instead of view:pre:change but view:pre:change gives you the option of including a cancel button to allow the user to keep web chat open.

vs

>Using "view:change" allows us to display a custom panel after the web chat window has been opened; with "view:pre:change" web chat will not be open yet and you can't show a custom panel if web chat is closed.

If the web chat needs to be open then is the former statement not true?

Reply:
I'm going to just pull out all the discussion about the pre:change event from here. It is just confusing here.